crates/bevy_animation/src/graph.rs

```diff
@@ -0,0 +1,467 @@
+}
+
+impl<
+        'gr,
+        't,
+        'ac,
+        'p,
+        'ppq,
+        'ppa,
+        'ppb,
+        'ppc,
+        'cq,
+        'ca,
+        'cb,
+        'cc,
+        'nq,
+        'na,
+        'nb,
+        'nc,
+        'tq,
+        'ta,
+        'tb,
+        'tc,
+        'mwq,
+        'mwa,
+        'mwb,
+        'mwc,
+    >
+    AnimationContext<
+        'gr,
+        't,
+        'ac,
+        'p,
+        'ppq,
+        'ppa,
+        'ppb,
+        'ppc,
+        'cq,
+        'ca,
+        'cb,
+        'cc,
+        'nq,
+        'na,
+        'nb,
+        'nc,
+        'tq,
+        'ta,
+        'tb,
+        'tc,
+        'mwq,
+        'mwa,
+        'mwb,
+        'mwc,
+    >
+{
+    fn evaluate(&mut self) {
+        self.evaluate_node(self.animation_graph.root, AnimationWeight(1.0))
+    }
+
+    fn evaluate_node(&mut self, node_index: AnimationNodeIndex, weight: AnimationWeight) {
+        let kids: Vec<_> = self
+            .animation_graph
+            .graph
+            .neighbors_directed(*node_index, Direction::Outgoing)
+            .collect();
+        for &kid in &kids {
+            // FIXME: Is `weight * kid_edge.weight()` right?
+            self.evaluate_node(kid.into(), weight * self.animation_graph.graph[kid].weight);
+        }
+
+        let node = &mut self.animation_graph.graph[*node_index];
+        if let Some(ref mut playing_animation) = node.animation {
+            crate::apply_animation(
+                *weight,
+                playing_animation,
+                node.paused,
+                self.root_entity,
+                self.time,
+                self.animation_clips,
+                self.names,
+                self.transforms,
+                self.morph_weights,
+                self.parent,
+                self.parents,
+                self.children,
+            )
+        }
+    }
+}
+
+impl From<NodeIndex<u32>> for AnimationNodeIndex {
+    fn from(value: NodeIndex<u32>) -> Self {
+        Self(value)
+    }
+}
+
+impl Into<NodeIndex<u32>> for AnimationNodeIndex {
+    fn into(self) -> NodeIndex<u32> {
+        *self
+    }
+}
+
+impl From<f32> for AnimationWeight {
+    fn from(value: f32) -> Self {
+        Self(value)
+    }
+}
+
+impl Mul<AnimationWeight> for AnimationWeight {
+    type Output = AnimationWeight;
+
+    fn mul(self, rhs: AnimationWeight) -> Self::Output {
+        Self(self.0 * rhs.0)
+    }
+}
+
+/// A system that updates all playing animations in animation graphs.
+pub fn evaluate_animation_graphs(
+    time: Res<Time>,
+    animation_clips: Res<Assets<AnimationClip>>,
+    children: Query<&Children>,
+    names: Query<&Name>,
+    transforms: Query<&mut Transform>,
+    morph_weights: Query<&mut MorphWeights>,
+    parents: Query<(Has<AnimationPlayer>, Has<AnimationGraph>, Option<&Parent>)>,
+    mut animation_graphs: Query<(Entity, Option<&Parent>, &mut AnimationGraph)>,
```

Comment:
This seems to drop the parallelization that is currently supported, and may be a significant performance regression for scenes with a large number of animated rigs.

Reply:
Is there any reason we cannot just use `.par_iter_mut().for_each` here? Seems like it would work.

Reply:
It does look like `par_iter` would bring this in line with how things are currently dispatched in `animation_player`. Same exact query more or less.